prover/crates/bin/witness_generator/src/leaf_aggregation.rs

```diff
@@ -255,29 +245,109 @@ pub async fn prepare_leaf_aggregation_job(
         circuit_id: metadata.circuit_id,
         block_number: metadata.block_number,
         closed_form_inputs: closed_form_input,
-        proofs: base_proofs,
+        proofs_ids: metadata.prover_job_ids_for_proofs,
         base_vk,
         leaf_params,
     })
 }
 
+// The number of leaf circuits to be processed in parallel.
+// Allows us to easily adjust the balance between processing speed and memory usage.
+// Each circuit requires downloading RECURSION_ARITY (32) proofs, each of which can be roughly estimated at 1 MB.
+// So processing of MAX_IN_FLIGHT_CIRCUITS == 100  should use ~3200 MB of RAM + some overhead during serialization.
+const MAX_IN_FLIGHT_CIRCUITS: usize = 100;
```

Comment:
Nit: We now have a config variable for that, would it make sense to reuse it here?

Reply:
We already have them separate in our infra, this would be defined on the per-deployment level (same as `group_id`).
And in the future, the permits will be memory-based, so a heterogenous load would be handled automatically.

Reply:
Made using config. 500 is suitable for LWG/NWG too, but I would prefer 100, since I don't expect a big difference in speed